internal/gen/gen.go

```diff
@@ -2,136 +2,327 @@
+	result := make([]string, 0)
+	switch t := s.Type.(type) {
+	case string:
+		return []string{t}, nil
+	case []any:
+		for _, v := range t {
+			if v == "null" {
+				s.Nullable = true
+			} else {
+				result = append(result, fmt.Sprintf("%v", v))
+			}
+		}
+	case nil:
+	default:
+		return nil, fmt.Errorf("unknown type %T", s.Type)
 	}
 
-	out := make(map[string]types.UserConfigSchema, len(r))
+	// Golang supports null for required fields only
+	if s.isRequired && s.Nullable {
+		result = append(result, "null")
+	}
 
-	for _, v := range r {
-		cv, err := convert.UserConfigSchema(v.UserConfigSchema)
-		if err != nil {
-			return err
-		}
+	for _, v := range s.AnyOf {
+		if v.Type != nil {
+			vType, err := normalizeType(v)
+			if err != nil {
+				return nil, err
+			}
 
-		out[v.EndpointType] = *cv
+			result = append(result, vType...)
+		}
 	}
 
-	result[types.KeyIntegrationEndpointTypes] = out
+	if len(result) == 0 {
+		result = append(result, "string")
+	}
 
-	return nil
+	return distinct(result), nil
 }
 
-// setup sets up the generation process.
-func setup(l *util.Logger, e util.EnvMap, c *aiven.Client) {
-	logger = l
-	env = e
-	client = c
+func formatValue(t string, v any) any {
+	s := fmt.Sprintf("%v", v)
+	if v == nil || s == "" {
+		return nil
+	}
+
+	switch t {
+	case "integer":
+		i, ok := v.(float64)
+		if ok && isSafeInt(i) {
+			return fmt.Sprintf("%d", int(i))
+		}
+		return nil
+	case "number":
+		return fmt.Sprintf("%.1f", v)
+	case "boolean", "array", "object":
+		return v
+	}
 
-	result = types.GenerationResult{}
+	return s
 }
 
 // Run executes the generation process.
-func Run(
-	ctx context.Context,
-	logger *util.Logger,
-	env util.EnvMap,
-	client *aiven.Client,
-) (types.GenerationResult, error) {
-	setup(logger, env, client)
+func Run(fileNames ...string) (types.GenerationResult, error) {
+	result := make(types.GenerationResult)
+	for _, fileName := range fileNames {
+		subResult, err := fromFile(fileName)
+		if err != nil {
+			return nil, err
+		}
 
-	g, ctx := errgroup.WithContext(ctx)
+		for kind, value := range subResult {
+			if len(value) == 0 {
+				continue
+			}
 
-	g.Go(func() error { return serviceTypes(ctx) })
-	g.Go(func() error { return integrationTypes(ctx) })
-	g.Go(func() error { return integrationEndpointTypes(ctx) })
+			if _, ok := result[kind]; !ok {
+				result[kind] = value
+				continue
+			}
 
-	return result, g.Wait()
+			for k, v := range value {
+				result[kind][k] = v
+			}
+		}
+	}
+	return result, nil
+}
+
+// reUnderscore finds multiple underscores.
+var reUnderscore = regexp.MustCompile(`_+`)
+
+// toSnakeCase converts a string to snake case.
+// strcase fails with S3 like strings https://github.com/iancoleman/strcase/issues/42
+func toSnakeCase(s string) string {
+	var r string
+	for i, v := range s {
+		if i > 0 && v >= 'A' && v <= 'Z' {
+			r += "_"
+		}
+		r += string(v)
+	}
+	return strings.Trim(reUnderscore.ReplaceAllString(strings.ToLower(r), "_"), "_")
```

Comment:
Maybe we can use `huandu/xstrings`?

* `iancoleman/strcase`: https://go.dev/play/p/f_ZL_TSZItt "LogRouterS3BucketName" -> "log_router_s3_bucket_name"
* `toSnakeCase`: https://go.dev/play/p/2GsUjrr8z5t "LogRouterS3BucketName" -> "log_router_s3_bucket_name"
* `stoewer/go-strcase`: https://go.dev/play/p/i7XqmRlvmCU "LogRouterS3BucketName" -> "log_router_s3bucket_name"
* `huandu/xstrings`: https://go.dev/play/p/72i0T7DLUaJ "LogRouterS3BucketName" -> "log_router_s3_bucket_name"